tests/Tokenizer/Analyzer/AttributeAnalyzerTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Tokenizer\Analyzer;
+
+use PhpCsFixer\Tests\TestCase;
+use PhpCsFixer\Tokenizer\Analyzer\AttributeAnalyzer;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Tokenizer\Analyzer\AttributeAnalyzer
+ */
+final class AttributeAnalyzerTest extends TestCase
+{
+    /**
+     * Check it's not crashing for PHP lower than 8.0 as other tests run for PHP 8.0 only.
+     */
+    public function testNotAnAttribute(): void
+    {
+        $tokens = Tokens::fromCode('<?php class Foo { private $bar; }');
+        foreach ($tokens as $index => $token) {
+            static::assertFalse(AttributeAnalyzer::isAttribute($tokens, $index));
+        }
+    }
+
+    /**
+     * @requires     PHP 8.0
+     * @dataProvider provideIsAttributeCases
+     */
+    public function testIsAttribute(bool $isInAttribute, string $code): void
+    {
+        $tokens = Tokens::fromCode($code);
+
+        foreach ($tokens as $index => $token) {
+            if ($token->equals([T_STRING, 'Foo'])) {
+                if (isset($testedIndex)) {
+                    static::fail('Test is run against index of "Foo", multiple occurrences found.');
+                }
+                $testedIndex = $index;
+            }
+        }
+        if (!isset($testedIndex)) {
+            static::fail('Test is run against index of "Foo", not found in the code sample.');
+        }
+
+        static::assertSame($isInAttribute, AttributeAnalyzer::isAttribute($tokens, $testedIndex));
+    }
+
+    /**
+     * Test case requires to having "Foo" as it will be searched for to test its index.
+     */
+    public static function provideIsAttributeCases(): iterable
+    {
+        // Foo is not an attribute
+        foreach ([
```

Comment:
Should we include an explicit test for a string containing `'(Foo())'` inside an attribute parameter, eg `<?php #[Attr('Foo()')] class Bar {}`?